RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+# RFC: Amplify JS TypeScript Improvements
+
+Amplify JS is looking to improve our TypeScript support across the library to better serve our customers and provide a more intuitive & idiomatic developer experience. To this end, we are requesting your feedback on a variety of changes and improvements that will be available to you in our next major version release.
+
+This RFC is broken down into sections covering:
+
+- Library-wide TypeScript improvements
+- Changes to some of our core utilities such as `Hub` and Amplify configuration
+- Specific improvements to the `Auth`, `Storage`, and `API` categories and associated APIs
+
+We're also requesting feedback on any other TypeScript issues or pain points that you may have encountered not explicitly covered in this RFC.
+
+# Library-wide TypeScript Improvements
+
+Amplify JS will be making the following improvements to our TypeScript support. These improvements will be applied across the entire library, not just the categories highlighted below.
+
+- `strict` typings — We will be applying `strict` mode to the entire library to improve the usability of our types.
```

Comment:
maybe add a bit more on the benefit. "Amplify JS will just work with your application without any additional configurations if you have strict mode on."